File: service/scaladsl/sqs/client/src/main/scala/com/streetcontxt/lagom/internal/scaladsl/broker/sqs/SqsTopicFactory.scala
/*
 * Copyright (C) 2017 Quantify Labs Inc. <https://github.com/StreetContxt>
 */
package com.lightbend.lagom.internal.scaladsl.broker.sqs

import akka.actor.ActorSystem
import akka.stream.Materializer
import com.lightbend.lagom.internal.broker.sqs.SqsConfig
import com.lightbend.lagom.internal.scaladsl.api.broker.TopicFactory
import com.lightbend.lagom.scaladsl.api.Descriptor.TopicCall
import com.lightbend.lagom.scaladsl.api.{ServiceInfo, ServiceLocator}
import com.lightbend.lagom.scaladsl.api.broker.Topic

import scala.concurrent.ExecutionContext

/**
 * Factory for creating topics instances.
 */
private[lagom] class SqsTopicFactory(serviceInfo: ServiceInfo, system: ActorSystem, serviceLocator: ServiceLocator)(implicit materializer: Materializer, executionContext: ExecutionContext) extends TopicFactory {

  private val config = SqsConfig(system.settings.config)

  def create[Message](topicCall: TopicCall[Message]): Topic[Message] = {
    new ScaladslSqsTopic(config, topicCall, serviceInfo, system, serviceLocator)
  }
}

File: service/javadsl/sqs/client/src/main/java/com/streetcontxt/lagom/javadsl/broker/sqs/SqsTopicFactory.java
/*
 * Copyright (C) 2017 Quantify Labs Inc. <https://github.com/StreetContxt>
 */
package com.lightbend.lagom.javadsl.broker.sqs;

import akka.actor.ActorSystem;
import akka.stream.Materializer;

import com.lightbend.lagom.internal.broker.sqs.SqsConfig;
import com.lightbend.lagom.internal.broker.sqs.SqsConfig$;
import com.lightbend.lagom.internal.javadsl.api.broker.TopicFactory;
import com.lightbend.lagom.internal.javadsl.broker.sqs.JavadslSqsTopic;
import com.lightbend.lagom.javadsl.api.Descriptor;
import com.lightbend.lagom.javadsl.api.ServiceInfo;
import com.lightbend.lagom.javadsl.api.ServiceLocator;
import com.lightbend.lagom.javadsl.api.broker.Topic;
import scala.concurrent.ExecutionContext;

import javax.inject.Inject;

/**
 * Factory for creating topics instances.
 */
public class SqsTopicFactory implements TopicFactory {
    private final ServiceInfo serviceInfo;
    private final ActorSystem system;
    private final Materializer materializer;
    private final ExecutionContext executionContext;
    private final SqsConfig config;
    private final ServiceLocator serviceLocator;

    @Inject
    public SqsTopicFactory(ServiceInfo serviceInfo, ActorSystem system, Materializer materializer,
            ExecutionContext executionContext, ServiceLocator serviceLocator) {
        this.serviceInfo = serviceInfo;
        this.system = system;
        this.materializer = materializer;
        this.executionContext = executionContext;
        this.config = SqsConfig$.MODULE$.apply(system.settings().config());
        this.serviceLocator = serviceLocator;
    }

    @Override
    public <Message> Topic<Message> create(Descriptor.TopicCall<Message> topicCall) {
        return new JavadslSqsTopic<>(config, topicCall, serviceInfo, system, serviceLocator, materializer, executionContext);
    }
}
